lib/rules/selector-not-notation/index.js

```diff
@@ -110,10 +110,13 @@ const rule = (primary, _, context) => {
 						if (mustFix) return fixSimple(pseudo);
 					}
 
+					assert(pseudo.source && pseudo.source.end);
```

Comment:
We cannot use optional chaining since [Node.js 12 doesn't support it](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility). 😅 

It should be possible in the next major version. (maybe we will drop the Node.js 12 support)

https://github.com/stylelint/stylelint/blob/9802b9de2a40eb682df141ef2f121ade1631843c/package.json#L190-L192

Reply:
I have just tried `node --harmony-optional-chaining` (version 12.18) and `?.` works as intended.
Is it too cumbersome/impractical to add it?